sklearn/tests/test_discriminant_analysis.py

```diff
@@ -223,6 +225,60 @@ def test_lda_scaling():
                      'using covariance: %s' % solver)
 
 
+def test_lda_store_covariance():
+    # Test for slover 'lsqr' and 'eigen'
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+        # The deprecated test will be asserted once the deprecation
+        # is complete.
+
+        # Test the actual attribute:
+        clf = LinearDiscriminantAnalysis(solver=solver,
+                                         store_covariance=True).fit(X6, y6)
+        assert_true(hasattr(clf, 'covariance_'))
+
+        assert_array_almost_equal(
+            clf.covariance_[0],
+            np.array([0.422222, 0.088889])
+        )
+
+        assert_array_almost_equal(
+            clf.covariance_[1],
+            np.array([0.088889, 0.533333])
+        )
+
+    # Test for slover svd, the default is to not set the covariances_ attribute
+    clf = LinearDiscriminantAnalysis(solver='svd').fit(X, y)
+    assert_false(hasattr(clf, 'covariance_'))
+
+    # Test the actual attribute:
+    clf = LinearDiscriminantAnalysis(solver=solver,
+                                     store_covariance=True).fit(X6, y6)
+    assert_true(hasattr(clf, 'covariance_'))
+
+    assert_array_almost_equal(
+        clf.covariance_[0],
+        np.array([0.422222, 0.088889])
+    )
+
+    assert_array_almost_equal(
+        clf.covariance_[1],
+        np.array([0.088889, 0.533333])
+    )
+
+
+def test_lda_deprecation():
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+
+        # Test the deprecation
+        assert_warns(FutureWarning, clf.fit, X, y)
```

Comment:
Wait, I don't get it. Why is fit causing a `FutureWarning`? There should only be warnings if the user does something that will break in the future, i.e. access `covariance_` if they should not, or supply `store_covariances` instead of `store_covariance`.

Indeed, this should be `assert_no_warnings`.